src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php

```diff
@@ -373,17 +371,37 @@ private function isMaxDepthReached(array $attributesMetadata, $class, $attribute
     }
 
     /**
-     * Gets the cache key to use.
+     * Overwrite to update the cache key for the child.
+     *
+     * {@inheritdoc}
+     */
+    protected function createChildContext(array $parentContext, $attribute)
+    {
+        $context = parent::createChildContext($parentContext, $attribute);
+        // format is already included in the cache_key of the parent.
+        $context['cache_key'] = $this->getAttributeCacheKey('', $context);
+
+        return $context;
+    }
+
+    /**
+     * Build the cache key for the attributes cache.
+     *
+     * The key must be different for every option in the context that could change which attributes should be handled.
      *
      * @param string|null $format
      * @param array       $context
      *
      * @return bool|string
      */
-    private function getCacheKey($format, array $context)
+    private function getAttributeCacheKey($format, array $context)
     {
         try {
-            return md5($format.serialize($context));
+            return md5($format.serialize([
+                'context' => $context,
+                'ignored' => $this->ignoredAttributes,
+                'camelized' => $this->camelizedAttributes,
+            ]));
```

Comment:
i hope i caught all normalizer state that could influence which attributes are available and not too many.

things like circular reference or callbacks should not change which attributes are available i guess?

(serialize one array rather than concat several serialize result, assuming that gives better performance)